src/deprecate.ts

```diff
@@ -1,65 +1,23 @@
 import Vue, { PluginObject } from 'vue';
-
-function getOptions(component: Vue) {
-  if (!component.$vnode) {
-    return;
-  }
-
-  return (component.$vnode.componentOptions?.Ctor as any).options;
-}
-
-function checkComponent (component: Vue) {
-  const options = getOptions(component);
-  if (!options || !options.deprecated) {
-    return;
-  }
-
-  if (typeof options.deprecated === "string") {
-    printDeprecated(options.deprecated);
-  } else {
-    printDeprecated('Component ' + options.name + ' is deprecated');
-  }
-}
-
-function checkProps (component: Vue) {
-  const options = getOptions(component);
-  if (!options || !options.props || typeof options.props !== 'object') {
-    return;
-  }
-
-  const propsData = component.$options.propsData as any;
-  if (!propsData) {
-    return
-  }
-
-  Object.keys(options.props).forEach(key => {
-      const value = options.props[key];
-      if (!propsData[key] || !value.deprecated) {
-        return;
-      }
-
-      if (typeof value.deprecated === "string") {
-        printDeprecated(value.deprecated);
-      } else {
-        printDeprecated(`Property ${key} in component ${options.name} is deprecated`);
-      }
-    }
-  )
-}
-
-
-function printDeprecated (message: string) {
-  console.warn(`[DEPRECATED] ${message}`);
-}
+import { checkComponent } from '@/components';
+import { checkProps } from '@/properties';
 
 interface VueDeprecateOptions {
+  enabledOnProduction?: boolean;
+}
 
+const defaultOptions: VueDeprecateOptions = {
+  enabledOnProduction: false,
 }
 
 const VueDeprecate: PluginObject<VueDeprecateOptions> = {
-  install: function (vue: typeof Vue, options?: VueDeprecateOptions) {
+  install: function (vue: typeof Vue, options: VueDeprecateOptions = defaultOptions) {
     vue.mixin({
       created: function () {
+        if (process.env.NODE_ENV === 'production' && options && !options.enabledOnProduction) {
```

Comment:
Verify if options passed with some different property but not `enabledOnProduction`